File: Compiler/src/compiler/values/DecisionTreeVarValue.java
package compiler.values;

import compiler.Value;
import util.CompilationResult;
import util.DataType;
import util.JenaUtil;
import util.NamingManager;

import static util.JenaUtil.POAS_PREF;
import static util.JenaUtil.VAR_PRED;

public class DecisionTreeVarValue extends Value {

    public DecisionTreeVarValue(String value) {
        super(value);
    }

    @Override
    public DataType resultDataType() {
        return DataType.DECISION_TREE_VAR;
    }

    @Override
    public CompilationResult compile() {
        String resVarName = NamingManager.genVarName();
        return new CompilationResult(resVarName,
                JenaUtil.genTriple(resVarName, JenaUtil.genLink(POAS_PREF, VAR_PRED), JenaUtil.genStingVal(value)),
                "");
    }
}

File: Compiler/src/compiler/values/DecisionTreeVarValue.kt
package compiler.values

import compiler.Value
import util.CompilationResult
import util.DataType
import util.JenaUtil
import util.JenaUtil.POAS_PREF
import util.JenaUtil.VAR_PREDICATE
import util.JenaUtil.genTriple
import util.NamingManager

class DecisionTreeVarValue(value: String) : Value(value) {

    override fun resultDataType(): DataType = DataType.DecisionTreeVar

    override fun compile(): CompilationResult {
        val resVarName = NamingManager.genVarName()
        return CompilationResult(
            resVarName,
            listOf(genTriple(
                resVarName,
                JenaUtil.genLink(POAS_PREF, VAR_PREDICATE),
                JenaUtil.genStingVal(value)
            )),
            ""
        )
    }
}